CMakeLists.txt

```diff
@@ -203,54 +203,54 @@ set(TEST_ENVIRONMENT
         OPENSSL_CONF=${CMAKE_CURRENT_SOURCE_DIR}/test/engine.cnf
         )
 add_executable(test_digest test_digest.c)
-target_link_libraries(test_digest ${OPENSSL_CRYPTO_LIBRARY})
+target_link_libraries(test_digest ${OPENSSL_CRYPTO_LIBRARIES})
 add_test(NAME digest COMMAND test_digest)
 set_tests_properties(digest PROPERTIES ENVIRONMENT "${TEST_ENVIRONMENT}")
 
 add_executable(test_ciphers test_ciphers.c)
-target_link_libraries(test_ciphers ${OPENSSL_CRYPTO_LIBRARY})
+target_link_libraries(test_ciphers ${OPENSSL_CRYPTO_LIBRARIES})
 add_test(NAME ciphers COMMAND test_ciphers)
 set_tests_properties(ciphers PROPERTIES ENVIRONMENT "${TEST_ENVIRONMENT}")
 
 # test_curves is an internals testing program, it doesn't need a test env
 add_executable(test_curves test_curves.c)
-target_link_libraries(test_curves gost_core ${OPENSSL_CRYPTO_LIBRARY})
+target_link_libraries(test_curves gost_core)
 add_test(NAME curves COMMAND test_curves)
 
 add_executable(test_params test_params.c)
-target_link_libraries(test_params ${OPENSSL_CRYPTO_LIBRARY})
+target_link_libraries(test_params ${OPENSSL_CRYPTO_LIBRARIES})
 add_test(NAME parameters COMMAND test_params)
 set_tests_properties(parameters PROPERTIES ENVIRONMENT "${TEST_ENVIRONMENT}")
 
 add_executable(test_derive test_derive.c)
-target_link_libraries(test_derive ${OPENSSL_CRYPTO_LIBRARY})
+target_link_libraries(test_derive ${OPENSSL_CRYPTO_LIBRARIES})
 add_test(NAME derive COMMAND test_derive)
 set_tests_properties(derive PROPERTIES ENVIRONMENT "${TEST_ENVIRONMENT}")
 
 add_executable(test_sign test_sign.c)
-target_link_libraries(test_sign ${OPENSSL_CRYPTO_LIBRARY})
+target_link_libraries(test_sign ${OPENSSL_CRYPTO_LIBRARIES})
 add_test(NAME sign/verify COMMAND test_sign)
 set_tests_properties(sign/verify PROPERTIES ENVIRONMENT "${TEST_ENVIRONMENT}")
 
 add_executable(test_tls test_tls.c)
-target_link_libraries(test_tls ${OPENSSL_CRYPTO_LIBRARY} ${OPENSSL_SSL_LIBRARY})
+target_link_libraries(test_tls ${OPENSSL_SSL_LIBRARIES} ${OPENSSL_CRYPTO_LIBRARIES})
```

Comment:
Unfortunately, it seems that `find_package(OpenSSL ...)` still doesn't set up a dependency relationship between libssl and libcrypto, so it must still be done manually.

Reply:
This got resolved by moving to OpenSSL target names